_ingest-pipelines/processors/dot-expander.md

```diff
@@ -0,0 +1,303 @@
+
+The following example response confirms that the pipeline is working as expected:
+
+```json
+{
+  "docs": [
+    {
+      "doc": {
+        "_index": "testindex1",
+        "_id": "1",
+        "_source": {
+          "field": "city, state"
+        },
+        "_ingest": {
+          "timestamp": "2023-11-17T23:49:27.597933805Z"
+        }
+      }
+    }
+  ]
+}
+```
+
+### Step 3: Ingest a document
+
+The following query ingests a document into an index named `testindex1`:
+
+```json
+PUT testindex1/_doc/1?pipeline=dot-expander
+{
+  "field": "Denver, CO"
+}
+```
+{% include copy-curl.html %}
+
+### Step 4 (Optional): Retrieve the document
+
+To retrieve the document, run the following query:
+
+```json
+GET testindex1/_doc/1
+```
+{% include copy-curl.html %}
+
+#### Response
+
+The following response confirms the document was indexed:
+
+```json
+{
+  "_index": "testindex1",
+  "_id": "1",
+  "_version": 58,
+  "_seq_no": 57,
+  "_primary_term": 30,
+  "found": true,
+  "_source": {
+    "field": "Denver, CO"
+  }
+}
+```
+
+## Nested fields
+
+The `dot_expander` processor consolidates the `user.address.city` and `user.address.state` fields by merging with an existing `address`, `city`, and `state` field nested under `user`. If the field is a scalar value, then it will turn that field into an array. Take for example the following document:
+
+```json
+{
+    "dot_expander": {
+        "field": "user.address.city",
+        "override": true
+    }
+}
+```
+
+With `override:false` (or unset), the dot-pathed top-level field gets appended to the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": ["Denver", "Boulder"],
+            "state": "CO"
+        }
+    }
+}
+```
+
+With `override:true`, the dot-pathed top-level field value overwrites the target object node, so it transforms into:
+
+```json
+{
+   "user": {
+        "address": {
+            "city": "Boulder",
+            "state": "CO"
+        }
+    }
+}
+```
+
+If the field value is set to a wildcard `*`, the processor expands all top-level dotted field names, for example:  
+
+```json
+{
+    "dot_expander": {
+        "field": "*"
+    }
+}
+```
+
+Take for example the following document:
+
+
+```json
+{
+    "user.address.city": "Denver",
+    "user.address.state": "CO"
+}
+```
+
+The `dot_expander` processor transforms that document into the following fields, all nested under `user`:
+
+```json
+{
+    "user": {
+        "address": {
+            "city": "Denver"
+        }
+    },
+    "user": {
+        "address": {
+            "state": "CO"
+        }
+    }
+}
+```
+
+If a field is nested within a structure without dots, you can use the `path` parameter to traverse the non-dotted structure. For example, if you have the field `user.address`, you can use the `dot_expander processor` to expand it into an object field named `user` with a nested field named `address`, as shown in the following example: 
+
+```json
```

Comment:
Please specify the actual field names. Wildcards are not supported. Also, this is better to have as a complete request, not part of a document, so users can run the examples.

Reply:
@msfroh Please see the preceding and following comments. Will you or your team provide a relevant OpenSearch example?